### PR TITLE
Fix & improve some examples

### LINKED ISSUE #2440: [BUG] Broken button '4' in inlinekeyboard2.py example
When the inline button '4' in the inlinekeyboard2.py example is clicked, it sends the same button as an option in the keyboard menu. Thus, when the user clicks it again, Telegram raises an error saying new message content and reply markup is same. This can confuse a user.

### Steps to reproduce
1. Run inlinekeyboard2.py, click `/start`.

2. Click button '1'.

3. Click button '4'.

4.  Click button '4' again.
5.  Error raised from Telegram.

Potential fix:
Change which button is offered in the keyboard menu.

### PR DESCRIPTION
**Pull request by harshil21**
- closes #2440 
- some typo fixes
- Handle empty inline queries without raising errors.
- remove unnecessary if condition in passportbot.py
- use `ReplyKeyboardRemove()` when conversation is over.
- use `ForceReply` in an example.

### CODE REVIEW

File: examples/persistentconversationbot.py [hunk 1]
```diff
@@ -77,18 +77,19 @@ def regular_choice(update: Update, context: CallbackContext) -> int:
     context.user_data['choice'] = text
     if context.user_data.get(text):
         reply_text = (
-            f'Your {text}, I already know the following about that: {context.user_data[text]}'
+            f'Your {text}? I already know the following about that: {context.user_data[text]}'
         )
     else:
         reply_text = f'Your {text}? Yes, I would love to hear about that!'
-    update.message.reply_text(reply_text)
+    update.message.reply_text(reply_text, reply_markup=ForceReply(selective=True))
```
Comment (Bibo-Joshi) [inline examples/persistentconversationbot.py:84]: Mh, if people ask for examples on how to use `ForceReply` I feel like it's not a good idea to have them run a conversationbot … Could you mind moving this to `echobot.py`? You can have the `start` message mention the user, which should work with `selective=True`

File: examples/persistentconversationbot.py [hunk 2]
```diff
@@ -77,18 +77,19 @@ def regular_choice(update: Update, context: CallbackContext) -> int:
...
 
     return TYPING_REPLY
 
 
 def custom_choice(update: Update, _: CallbackContext) -> int:
     update.message.reply_text(
-        'Alright, please send me the category first, ' 'for example "Most impressive skill"'
+        'Alright, please send me the category first, for example "Most impressive skill"',
```
Comment (Bibo-Joshi) [inline examples/persistentconversationbot.py:91]: same

**Review by Bibo-Joshi — CHANGES_REQUESTED** [review]
Thanks for the improvements! Left a comment. Also you mentioned:

> * Handle empty inline queries without raising errors.

but I don't see that? If you want to add that, maybe an idea would be to use `switch_pm_text` to display a help message similarly to what rulesbot does. That feature is also little known.

**Comment by harshil21** [comment]
> but I don't see that? 

Oops, I guess I discarded those changes by mistake. It was `if query == ""; return`. The exception raised when it's empty is 
`Can't parse inline query result: input message content is not specified`.

> maybe an idea would be to use `switch_pm_text` to display a help message similarly to what rulesbot does. 

So should I also put in `switch_pm_{text, parameter}`, along with the fix above?

File: examples/persistentconversationbot.py (hunk 1 above)
Comment (harshil21) [reply examples/persistentconversationbot.py:84]: Alright. I included this here cause it felt the like the most natural place to put it.

**Comment by Bibo-Joshi** [comment]
> > but I don't see that?
> 
> Oops, I guess I discarded those changes by mistake. It was `if query == ""; return`. The exception raised when it's empty is
> `Can't parse inline query result: input message content is not specified`.

Jup, makes sense :)

> > maybe an idea would be to use `switch_pm_text` to display a help message similarly to what rulesbot does.
> 
> So should I also put in `switch_pm_{text, parameter}`, along with the fix above?

If you like to. Not important, just an idea ;)

**Pull request merged**